### PR TITLE
Trim down supported file extensions on Windows and Linux

### PR DESCRIPTION
**Pull request by ccordoba12**
- Spyder is basically a Python based editor, so there's no need to set it as the default for a lot of extensions we don't really support.
- See https://github.com/spyder-ide/spyder/issues/22389 for the problems the previous associations can introduce.

Fixes spyder-ide/spyder#22389.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@mrclary, could you take a look at this one? Thanks!

**Comment by mrclary** [comment]
Let me take a look at this. I agree that Spyder should not be the default application, but it should be in the "Open with..." context menu and these files should be openable from a Spyder open-file dialogue.

File: recipe/spyder-menu.json [hunk 1]
```diff
@@ -19,62 +19,12 @@
...
-                        ".po",
-                        ".pot",
-                        ".pro",
-                        ".properties",
-                        ".pxd",
-                        ".pxi",
                         ".py",
+                        ".pyi",
                         ".pyw",
                         ".pyx",
```
Comment (mrclary) [inline recipe/spyder-menu.json:27]:
```suggestion
                        ".pyx"
```
Final list item with "," results in `JSONDecodeError`

File: recipe/spyder-menu.json [hunk 2]
```diff
@@ -86,11 +36,6 @@
                     "StartupWMClass": "Spyder",
                     "MimeType": [
                         "text/x-python",
```
Comment (mrclary) [inline recipe/spyder-menu.json:38]:
```suggestion
                        "text/x-python"
```
Final list item with "," results in `JSONDecodeError`

File: recipe/spyder-menu.json [hunk 3]
```diff
@@ -109,103 +54,7 @@
...
-                                "public.nasm-assembly-source",
-                                "public.objective-c-plus-plus-source",
-                                "public.objective-c-source",
-                                "public.opencl-source",
-                                "public.pascal-source",
-                                "public.patch-file",
-                                "public.perl-script",
-                                "public.php-script",
-                                "public.plain-text",
                                 "public.python-script",
```
Comment (mrclary) [inline recipe/spyder-menu.json:57]:
```suggestion
                                "public.python-script"
```
Final list item with "," results in `JSONDecodeError`

**Review by mrclary — CHANGES_REQUESTED** [review]
@ccordoba12, I still need to test this PR, but at least the trailing commas in lists need to be removed, otherwise `JSONDecodeError` results.

File: recipe/spyder-menu.json (hunk 3 above)
Comment (mrclary) [inline recipe/spyder-menu.json:57]:
```suggestion
                                "public.plain-text",
                                "public.python-script"
```

So, I would retain at least plain-text for macOS. We could retain them all, or add them at the request of users. On macOS, these do not make them default. The absence of them results in the application not included in "Recommended Applications" for opening the file.

Screenshot 2024-08-27 at 9 34 57 PM

**Comment by ccordoba12** [comment]
@mrclary, this is ready for another review.

**Review by mrclary — APPROVED** [review]
Looks good to me; thanks @ccordoba12

**Pull request merged**